pam/pam_test/module-transaction-dummy_test.go

```diff
@@ -0,0 +1,854 @@
+package pam_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/msteinert/pam"
+	"github.com/stretchr/testify/require"
+)
+
+func ptrValue[T any](value T) *T {
+	return &value
+}
+
+func bytesPointerDecoder(ptr pam.BinaryPointer) ([]byte, error) {
+	if ptr == nil {
+		return nil, nil
+	}
+	return *(*[]byte)(ptr), nil
+}
+
+func TestSetGetItem(t *testing.T) {
+	t.Parallel()
+	t.Cleanup(MaybeDoLeakCheck)
+
+	tests := map[string]struct {
+		item  pam.Item
+		value *string
+
+		wantValue    *string
+		wantGetError error
+		wantSetError error
+	}{
+		"Set user": {
+			item:  pam.User,
+			value: ptrValue("an user"),
+		},
+
+		"Returns empty when getting an unset user": {
+			item:      pam.User,
+			wantValue: ptrValue(""),
+		},
+
+		"Setting and getting an user": {
+			item:      pam.User,
+			value:     ptrValue("the-user"),
+			wantValue: ptrValue("the-user"),
+		},
+
+		// Error cases
+		"Error when setting invalid item": {
+			item:         pam.Item(-1),
+			value:        ptrValue("some value"),
+			wantSetError: pam.ErrBadItem,
+		},
+
+		"Error when getting invalid item": {
+			item:         pam.Item(-1),
+			wantGetError: pam.ErrBadItem,
+			wantValue:    ptrValue(""),
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			t.Cleanup(MaybeDoLeakCheck)
+
+			// These test inputs can not be all unset
+			if tc.value == nil && tc.wantValue == nil {
+				t.Fatalf("Setup: some test inputs were nil, but they must not be")
+			}
+
+			tx := NewModuleTransactionDummy(nil)
+
+			if tc.value != nil {
+				err := tx.SetItem(tc.item, *tc.value)
+				require.ErrorIs(t, err, tc.wantSetError)
+			}
+
+			if tc.wantValue != nil {
+				value, err := tx.GetItem(tc.item)
+				require.Equal(t, value, *tc.wantValue)
+				require.ErrorIs(t, err, tc.wantGetError)
+			}
+		})
+	}
+}
+
+func TestSetPutEnv(t *testing.T) {
```

Comment:
So, here, you are testing your mock, so that it’s complete from a functional perspective to your interface. However, no code in this project is using the env part, so why testing it? I would understand if this mock was part of the `pam` project and you offer to other to test with it, but this is not the case here.

Do you forsee them being used in the next PR?
Similar on other tested methods.

Reply:
As per environ variables I'm not using them here nor in the next branches, but since that's a method that needs to be implemented, I've just added it for completion. It may be needed in future.

Instead most of other methods provided by the mock will be used by other branches.

So, some of this code could be removed, and replaced by dummy methods, but they were simple enough that adding them wasn't a big deal (at least so I thought).

Reply:
Fine. I'll avoid it next time, at this point I guess since it's already written and tested code is better not to trash it, right?